FileSystem/FileMapperMocker.js

```diff
@@ -10,28 +25,490 @@ var USE_FILEMAPPER_MOCKER = true;
+  var root = this.getBaseDriveFolder();
+  // Find folder containing local file
+  for (var i = 0; i < pathSplit.length - 1; i++) {
+    var folderIterator = root.getFoldersByName(pathSplit[i]);
+    if (!folderIterator.hasNext()) {
+      throw new Error('File not found');
+    }
+    root = folderIterator.next();
+  }
+  // Obtain the file
+  var fileIterator = root.getFilesByName(pathSplit[pathSplit.length - 1]);
+  if (!fileIterator.hasNext()) {
+    throw new Error('File not found');
+  }
+  // Return file id
+  return fileIterator.next().getId();
+}
+
+/**
+ * Get Folder API
+ * Gets the folder id of a local folder
+ * @param {string} localPath Local filepath of the folder
+ * @return {string} Folder id of the local folder
+ */
+function getFolderId(localPath) {
+  // Get platform agnostic relative path split
+  var pathSplit = this.getRelativePathSplit(localPath);
+  // Get base directory
+  var root = this.getBaseDriveFolder();
+  // Find the directory
+  for (var i = 0; i < pathSplit.length; i++) {
+    var folderIterator = root.getFoldersByName(pathSplit[i]);
+    if (!folderIterator.hasNext()) {
+      throw new Error('Folder not found');
+    }
+    root = folderIterator.next();
+  }
+  return root.getId();
+}
+
+/**
+ * Has File API
+ * Checks if the local file exists
+ * @param {string} localPath Local filepath of the file
+ * @return {boolean} true if the local file exists
+ */
+function hasFile(localPath) {
+  // Get platform agnostic relative path split
+  var pathSplit = this.getRelativePathSplit(localPath);
+  // Get base directory
+  var root = this.getBaseDriveFolder();
+  // Search for directory containing the file
+  for (var i = 0; i < pathSplit.length - 1; i++) {
+    var folderIterator = root.getFoldersByName(pathSplit[i]);
+    if (!folderIterator.hasNext()) {
+      return false;
+    }
+    root = folderIterator.next();
+  }
+  // Check if file exists
+  var fileIterator = root.getFilesByName(pathSplit[pathSplit.length - 1]);
+  if (!fileIterator.hasNext()) {
+    return false;
+  }
+  return true;
+}
+
+/**
+ * Has Folder API
+ * Checks if the local folder exists
+ * @param {string} localPath Local filepath of the folder
+ * @return {boolean} true if the local folder exists
+ */
+function hasFolder(localPath) {
+  // Get platform agnostic relative path split
+  var pathSplit = this.getRelativePathSplit(localPath);
+  // Get base directory
+  var root = this.getBaseDriveFolder();
+  // Search for folder
+  for (var i = 0; i < pathSplit.length; i++) {
+    var folderIterator = root.getFoldersByName(pathSplit[i]);
+    if (!folderIterator.hasNext()) {
+      return false;
+    }
+    root = folderIterator.next();
+  }
+  return true;
+}
+
+/**
+ * Delete File API
+ * Deletes local file if it exists
+ * @param {string} localPath Local filepath of the file
+ */
+function deleteFile2(localPath) {
+  try {
+    var id = this.getFileId(localPath);
+    DriveApp.getFileById(id).setTrashed(true);
+  } catch (e) {
```

Comment:
Should we be catching the error? we should prompt / inform the user if something went wrong?

Reply:
Fixed